src/quark-core/src/QuarkNonceManager.sol

```diff
@@ -19,7 +19,7 @@ contract QuarkNonceManager {
     error InvalidNonce(address wallet, bytes32 nonce);
     error InvalidSubmissionToken(address wallet, bytes32 nonce, bytes32 submissionToken);
 
-    event NonceSubmitted(address wallet, bytes32 nonce, bytes32 submissionToken);
+    event NonceSubmitted(address indexed wallet, bytes32 indexed nonce, bytes32 submissionToken);
```

Comment:
I also just addded an index on `nonce` since it could be useful for replayable operations.